lib/ClosureF.hs

```diff
@@ -66,17 +66,19 @@ data Expr t e =
    -- Java
    | JNew ClassName [Expr t e]
    | JMethod (Either ClassName (Expr t e)) MethodName [Expr t e] ClassName
-   | JField  (Either ClassName (Expr t e)) FieldName ClassName
+   | JField  (Either ClassName (Expr t e)) FieldName (Type t)
    | PolyList [Expr t e] (Type t)
    | JProxyCall (Expr t e) (Type t)
    | SeqExprs [Expr t e]
    | Data S.RecFlag [DataBind t] (Expr t e)
    | Constr (Constructor t) [Expr t e]
    | Case (Expr t e) [Alt t e]
+   | CaseCast (Expr t e) (Constructor t)
```

Comment:
While entering a pattern, cast is needed.
Like

```
case x of
    Cons  y z -> case y of ...
```

Here `y` will become `JField (CaseCast x "Cons") "field1"`, namely `(Cons)x.field1` in java

I have thought about not adding `CaseCast` but have not figured out a better alternative way. Any suggestions?